src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -810,6 +810,24 @@ private static void EmitNumericConversion(this ILGenerator il, Type typeFrom, Ty
                     il.Emit(OpCodes.Conv_R_Un);
                 il.Emit(OpCodes.Conv_R8);
             }
+            else if (typeTo == typeof(Decimal))
+            {
+                // NB: TypeUtils.IsImplicitNumericConversion makes the promise that implicit conversions
+                //     from various integral types and char to decimal are possible. Coalesce allows the
+                //     conversion lambda to be omitted in these cases, so we have to handle this case in
+                //     here as well, by using the op_Implicit operator implementation on System.Decimal
+                //     because there are no opcodes for System.Decimal.
+
+                Debug.Assert(typeFrom != typeTo);
+
+                MethodInfo method = typeof(Decimal).GetMethod("op_Implicit", new[] { typeFrom });
```

Comment:
If we take PR https://github.com/dotnet/corefx/pull/11334 for caching of reflection members, we may want to consider adding this handful of `op_Implicit` methods to the cache as well and switch on the `TypeCode` of `typeFrom` to find the method we need (with `default` throwing an exception for code that should be unreachable).